### PR TITLE
fix: remove unused devDependencies, switch to c8, remove make-dir

### PR DESCRIPTION
**Pull request by XhmikosR**
Also:

1. replace make-dir and lodash packages with native
2. replace nyc with c8

Notes:

1. Opening for discussion. I know ideally I should have split this, but time :)
2. There's a TODO that should be addressed:
  https://github.com/netlify/netlify-plugin-edge-handlers/blob/093e0eb2097d5e06e3b1f7f661b6209f49d06e4f/src/node-compat/globals.js#L7-L9
3. There were plenty of devDependencies present that are in `@netlify/eslint-config-node` hence why I dropped them
4. `rollup-plugin-node-polyfills` should be replaced because it's severely outdated
5. `del` can be replaced later when Node.js 12 support is dropped (or now if you are willing to workaround the deprecated `rmdir`)

I can try splitting some of these changes later if you want.

---

🎉 Thanks for sending this pull request! 🎉

Please make sure the title is clear and descriptive.

If you are fixing a typo or documentation, please skip these instructions.

Otherwise please fill in the sections below.

**Which problem is this pull request solving?**

Example: I'm always frustrated when [...]

**List other issues or pull requests related to this problem**

Example: This fixes #5012

**Describe the solution you've chosen**

Example: I've fixed this by [...]

**Describe alternatives you've considered**

Example: Another solution would be [...]

**Checklist**

Please add a `x` inside each checkbox:

- [x] The status checks are successful (continuous integration). Those can be seen below.

### CODE REVIEW

File: test/fixtures/integration-test/netlify/edge-handlers/hello_world.js [hunk 1]
```diff
@@ -1,13 +1,11 @@
-// eslint-disable-next-line you-dont-need-lodash-underscore/concat
-import { concat } from 'lodash'
```
Comment (XhmikosR) [inline test/fixtures/integration-test/netlify/edge-handlers/hello_world.js:2]: Unsure if this was intentional for testing purposes. If so I can revert the changes.

File: package.json [hunk 2]
```diff
@@ -43,24 +43,34 @@
...
-    "typescript": "^4.1.5"
+    "typescript": "^4.5.4"
+  },
+  "devDependencies": {
+    "@netlify/build": "^26.1.0",
+    "@netlify/eslint-config-node": "^4.0.6",
+    "ava": "^3.15.0",
+    "c8": "^7.11.0",
+    "date-fns": "^2.28.0",
+    "husky": "^4.3.8",
```
Comment (XhmikosR) [inline package.json:70]: I kept husky although it's already included in `@netlify/eslint-config-node`. Not sure what's your policy is.

File: package.json [hunk 3]
```diff
@@ -43,24 +43,34 @@
...
+    "is-plain-obj": "^3.0.0",
+    "npm-run-all": "^4.1.5",
```
Comment (XhmikosR) [inline package.json:72]: This is also included in `@netlify/eslint-config-node`

File: package.json [hunk 4]
```diff
@@ -43,24 +43,34 @@
...
-    "node-fetch": "^2.6.1",
+    "node-fetch": "^2.6.6",
     "path-type": "^4.0.0",
     "process-es6": "^0.11.6",
-    "rollup": "^2.23.1",
+    "rollup": "^2.63.0",
     "rollup-plugin-node-polyfills": "^0.2.1",
     "rollup-plugin-terser": "^7.0.2",
-    "typescript": "^4.1.5"
+    "typescript": "^4.5.4"
```
Comment (XhmikosR) [inline package.json:62]: Is typescript really used? It's a big dependency as you probably know, and it's probably intended but I thought I'd ask just to be safe.

Comment (XhmikosR) [reply package.json]: @erezrokah I tested the package without typescript as a dependency and tests pass., so either TS isn't covered by the current tests or it's an unused dependency...

**Comment by XhmikosR** [comment]
BTW it seems that this fixes coverage in this repo because I guess nyc didn't include any files.

I'd still be interested to see if typescript is needed as a dependency. If not, it's a huge win. If it is, I guess the test suite needs to cover it too.

File: package.json (hunk 4 above)
Comment (erezrokah) [reply package.json]: Looks like it was introduced in https://github.com/netlify/netlify-plugin-edge-handlers/commit/1b7ed5e1c7d722bea1cfa25b565274f909c0d763

Doesn't look like we need it

Comment (XhmikosR) [reply package.json]: The only `ts` reference I can find is https://github.com/netlify/netlify-plugin-edge-handlers/blob/2c5c9a6837995282c89b78c8bccbff928eb18ca2/src/consts.js#L3

Comment (XhmikosR) [reply package.json]: Regardless, it'd be better to remove typescript in a separate PR, I just noticed it here and thought I'd ask about it.

I'll make a PR after this lands.

File: package.json (hunk 3 above)
Comment (erezrokah) [reply package.json:71]: Yes! I think you can remove it

**Comment by XhmikosR** [comment]
This is ready from my side @erezrokah :)

**Review by erezrokah — APPROVED** [review]

**Comment by XhmikosR** [comment]
@erezrokah feel free to tackle the remaining TODO from the [OP](https://github.com/netlify/netlify-plugin-edge-handlers/pull/829#issue-1095061347) later :)

**Pull request merged**